kolibri/plugins/learn/assets/src/state/mutations/index.js

```diff
@@ -35,12 +35,22 @@ export default {
   LEARN_SET_MEMBERSHIPS(state, memberships) {
     state.learnAppState.memberships = memberships;
   },
-  SET_TOPIC_PROGRESS(state, progressArray) {
+  SET_NODE_PROGRESS(state, progressArray) {
     progressArray.forEach(progress => {
-      const topic = state.pageState.contents.find(subtopic => subtopic.id === progress.pk);
-      if (topic) {
-        topic.progress = progress.progress_fraction;
+      const contentNode = state.pageState.contents.find(node => node.id === progress.pk);
+      if (contentNode) {
+        contentNode.progress = progress.progress_fraction;
       }
     });
   },
+  SET_RECOMMENDED_NODES_PROGRESS(state, progressArray) {
+    Object.keys(state.pageState).forEach(function(key) {
```

Comment:
Might want to make this explicit and use `['recommended', 'popular', 'nextSteps']` instead of reading the object keys of the `pageState`.

Reply:
I was thinking about that but wasn't sure about the general verbosity philosophy so I went with this, but will gladly update as per your suggestion.

Reply:
My main reason for explicitness is just in case there are other keys on the pageState that aren't one of these three.

Reply:
Yes, I understand completely, my thinking was that if another key was to be added to `pageState` it would be another set of _Recommended_ nodes which we'd want to annotate anyway - but I agree that listing them explicitly is definitely safer and someone will have to add it to the mutation manually if they see that it's not working rather than causing regression if the key added to `pageState` wasn't a list of nodes.